### PR TITLE
feat(mactaskbar): rebuild in Swift so the app can see and reach the menu bar

### PR DESCRIPTION
**Pull request by choisungwook**
# Goal

akbun-mactaskbar drew nothing on a full menu bar. The process ran, its three status items existed with positive coordinates, and none of them were on screen. This PR rebuilds the app on Swift and AppKit so it can set divider width in points, tell a visible item from one parked under the camera housing, and stay usable when its own icon cannot be drawn.

# Decisions

**[Important]** An item is visible only when it sits right of the camera housing, not when x is positive.
- On a notched display the menu bar splits in two and status items only land in the right half. What does not fit is placed under the housing with an ordinary positive x and drawn nowhere.
- NSScreen.auxiliaryTopRightArea reports that edge. It measured 956 on a 1728pt display, where four items between 878 and 955 were invisible while reporting positive coordinates.
- Every visibility answer now goes through one type, so the item list and the app's own icon check cannot disagree.

**[Important]** The app has to survive its own control icon being undrawable.
- macOS gives a new status item the leftmost free slot. On a full bar that slot is under the housing, so this app loses its only click target on exactly the machines it is built for.
- Nothing inside the app can fix that. Where an item goes is macOS's decision, and there is no way to make room or move somebody else's icon.
- So the app detects it and opens the item list window on launch instead of looking like it failed to start. Launching the app again reopens that window.

**[Important]** The global shortcut is a setting, because a conflict cannot be detected.
- Registering a combination another app already owns returns noErr and then never fires. There is no API to ask who holds what, so there is nothing to report.
- Three choices plus off, control-command-B by default. Changing it unregisters and re-registers on the spot, so a clash is something the user walks out of.
- Carbon over an NSEvent monitor because it needs no accessibility permission and consumes the key.

We left Electron.
- Tray exposes an image and a title and nothing else, so divider width had to be expressed as a run of spaces betting a space is about 4pt.
- There is no route from Electron to the housing geometry, so it can neither detect the failure above nor explain it.
- Cost accepted: a second toolchain for one app, and the pull request check moves from ubuntu to macOS.

Only expanded folds back on the timer.
- all is where icons get assigned, by holding Command and dragging them across a divider.
- That drag outlasts any sensible delay, so folding the bar mid-drag would make the one job needing that state impossible to finish.

# Implementation

**[Important]** NSStatusItem.length replaces the run of spaces. Measured on a 1728pt display, x and width of the app's three items in each state:

```text
                control        hidden divider   always-hidden
collapsed    895 w24          -2577 w3458      -6049 w3458
expanded     895 w24            867 w14        -2605 w3458
all          895 w24            867 w14          839 w14
```

- Narrow is 12pt, which macOS renders as 14. A full square would spend two icon slots on a bar that has none to spare.
- Wide is twice the screen width, floored at 2000. One width clears the edge, the second is headroom for a bar whose items already start left of the origin.
- All three states verified on the real menu bar by restoring each from UserDefaults and reading positions back through the accessibility API.

The bar scan reads AXExtrasMenuBar once per process.
- 1.26s for the full sweep against 37 items, where the osascript version took about 10s for 21.
- AXUIElementSetMessagingTimeout bounds each call at one second, and processes are visited through concurrentPerform. A wedged app costs one worker.
- The scan runs on a detached task. Blocking the main thread would freeze the menu bar the app is managing.

Control Center placeholders are filtered by width, not position.
- It lists one zero-sized entry per system extra that is switched off, 15 of them on the test machine, all at the top-left corner.
- Position cannot be the test: an item pushed off screen keeps its real width, and the app's own dividers are the widest things on the bar.

22 tests in MacTaskbarCore, which imports no AppKit. They cover the state to width mapping, the housing boundary against measured coordinates, the placeholder filter, release version arithmetic and the dmg architecture pick.

scripts/bundle.sh replaces electron-builder. It builds release, writes Info.plist with LSUIElement, ad-hoc signs because arm64 will not run an unsigned binary, and packs the dmg. Version moves from package.json to VERSION.

Not verified, because the machine was locked for the second half of the work: the rendered window, clicking the control icon, the right-click menu, the shortcut firing, and the auto-collapse timer. Window creation was confirmed from stderr with isVisible true at a real frame, and all three shortcut combinations register with status 0.

# Challenges

The first suspicion, that the empty Tray image was at fault, was wrong.
- A plain Electron app with one empty-image tray and one real-image tray showed neither. That ruled out rendering.
- Reading positions through the accessibility API found the items at x 785 and 831, behind the notch. The bar was simply full.

The window-opens-on-launch fix looked broken and was not.
- The accessibility API reported zero windows for the app, but stderr showed the window created with isVisible true at a real frame.
- macOS does not expose an accessory app's windows to the accessibility tree while the screen is locked.

Claiming AXExtrasMenuBar needed no filter was wrong.
- The first full scan returned 15 phantom rows from Control Center at x 0, which would have made the off-screen tally, the one number the window exists to show, wrong by 15.
- Their size is 0x0, so width became the discriminator and the decision record was corrected.

Issue #607

### CODE REVIEW

File: product/akbun-mactaskbar/workspace/Sources/MacTaskbarCore/Sections.swift [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  }
+
+  /// SF Symbol shown on the control item. The arrow points at what a click does:
+  /// reveal what sits to the left, or fold it all back up.
+  public var controlSymbol: String {
+    switch self {
+    case .collapsed: "chevron.left"
+    case .expanded: "chevron.left.2"
+    case .all: "chevron.right"
+    }
```
[the remark this answers is not in the document]
Comment (choisungwook) [reply product/akbun-mactaskbar/workspace/Sources/MacTaskbarCore/Sections.swift:35]: - 수락여부: 거부
- 이유: Swift 5.9부터 switch가 표현식이라 computed property에서 return 없이 컴파일된다. swift build가 경고 없이 통과하고 CI verify도 통과했다.

File: product/akbun-mactaskbar/workspace/Sources/akbun-mactaskbar/Updater.swift [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+
+  enum Failure: LocalizedError {
+    case badStatus(Int)
+    case noBundleInDmg
+
+    var errorDescription: String? {
+      switch self {
+      case .badStatus(let code): "GitHub returned status \(code)"
+      case .noBundleInDmg: "The dmg holds no app bundle"
+      }
```
[the remark this answers is not in the document]
Comment (choisungwook) [reply product/akbun-mactaskbar/workspace/Sources/akbun-mactaskbar/Updater.swift:55]: - 수락여부: 거부
- 이유: 위와 같다. Swift 5.9 switch 표현식이라 errorDescription도 return 없이 컴파일된다. 빌드와 CI 모두 통과했다.

File: product/akbun-mactaskbar/workspace/Sources/akbun-mactaskbar/Hotkey.swift [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    let id = EventHotKeyID(signature: OSType(0x414B_4254), id: 1)
+    RegisterEventHotKey(
+      UInt32(kVK_ANSI_B),
+      UInt32(controlKey | cmdKey),
+      id,
+      GetApplicationEventTarget(),
+      0,
+      &hotKeyRef
+    )
+  }
```
[the remark this answers is not in the document]
Comment (choisungwook) [reply product/akbun-mactaskbar/workspace/Sources/akbun-mactaskbar/Hotkey.swift]: - 수락여부: 수락
- 이유: InstallEventHandler와 RegisterEventHotKey의 status를 모두 확인하고 실패하면 창에 표시하도록 고쳤다. 단축키는 아이콘이 노치에 가렸을 때의 진입 경로라 조용한 실패가 특히 나쁘다. 다만 다른 앱과의 조합 충돌은 noErr을 돌려주므로 이 검사로 잡히지 않는다.

File: product/akbun-mactaskbar/workspace/Sources/akbun-mactaskbar/AppDelegate.swift [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+    sections = SectionController()
+    sections.onContextMenu = { [weak self] in self?.contextMenu() ?? NSMenu() }
+    model = ItemsModel(sections: sections)
+
+    Hotkey.register()
+    NotificationCenter.default.addObserver(
+      forName: .mactaskbarHotkey, object: nil, queue: .main
+    ) { [weak self] _ in
+      MainActor.assumeIsolated { self?.sections.cycle() }
+    }
```
[the remark this answers is not in the document]
Comment (choisungwook) [reply product/akbun-mactaskbar/workspace/Sources/akbun-mactaskbar/AppDelegate.swift:22]: - 수락여부: 수락
- 이유: 두 observer 모두 Task { @MainActor in }으로 바꿨다. SectionController가 이미 같은 방식을 쓰고 있어 코드가 일관되기도 한다.

File: knowledge/log.md [hunk 5]
```diff
@@ -2,6 +2,8 @@
 
 ## 2026-07-31
 
+* **Update**: [macOS 메뉴바 status item은 넓은 자리 차지로만 가릴 수 있고, 노치 뒤에서는 좌표가 있어도 그려지지 않는다](topics/macos-menubar-status-items.md)를 노치 제약과 네이티브 실측으로 갱신했다. akbun-mactaskbar를 Swift로 다시 만들면서 아이콘이 하우징 아래에 배치되어 보이지 않던 원인을 찾았고, Electron 전제로 쓰인 부분을 걷어냈다.
+
 * **Creation**: [PR body 형식의 기준은 pull request template 하나다](decisions/2026-07-pr-body-format-in-template.md) 결정 기록. PR body를 Decisions와 Implementation으로 바꾸면서 형식이 네 파일에 흩어져 있던 문제를 함께 정리한다.
 * **Creation**: [Electron 메뉴바 앱은 창 정리와 메뉴바 공간에서 조용히 실패한다](topics/electron-menubar-silent-failures.md) topic 기록. akbun-screenshot의 프리뷰 창이 화면 밖으로 밀려나던 버그를 고치면서 남긴다.
 * **Creation**: [macOS 메뉴바 status item은 넓은 자리 차지로만 가릴 수 있고, 목록은 프로세스 단위로만 읽을 수 있다](topics/macos-menubar-status-items.md) topic 작성. akbun-mactaskbar를 만들며 실측한 제약과 우회 방법을 남긴다.
```
[the remark this answers is not in the document]
Comment (choisungwook) [reply knowledge/log.md]: - 수락여부: 수락
- 이유: 생성과 갱신이 같은 날이라 Update 항목을 지우고 Creation 항목 하나로 합쳤다. 옛 제목을 가리키던 링크가 사라졌다.
